pymodbus/pdu/pdu.py

```diff
@@ -36,11 +35,14 @@ def __init__(self,
         self.bits: list[bool] = bits or []
         if not registers:
             registers = []
-        self.registers: list[int] = cast(list[int], registers)
-        for i, value in enumerate(registers):
+        self.registers: list[int] = []
+        # copy values so original sequence is not mutated
+        for value in registers:
             if isinstance(value, bytes):
-                self.registers[i] = int.from_bytes(value, byteorder="big")
-        self.count: int = count or len(registers)
+                value = int.from_bytes(value, byteorder="big")
+            self.registers.append(value)
```

Comment:
This is quite a costly operation, which is now done with every entry.

The old code fast copied the entire array and only update entries that contained bytes, ensuring a lot faster execution for well behaving calls.

Reply:
It's not as costly as it looks. `Sequence` does not have a `copy` method, so that was not an option. I could have used a slice `[:]` but since we have to iterate over the entire list to convert the bytes, we may as well only iterate _once_, An explicit copy would iterate once to get the elements and then a second time to convert the bytes.

I figured, at least this way there's only one iteration. When i get back to the office i could certainly do performance testing to see which is the quickest option.

However, the typing has now changed again so this entire MR is kind of pointless. It only mattered because you typed this as `Sequence` which implies immutability.

I do think, however, that if you were really concerned about performance, there would not be a loop inside this function to iterate through the entire list and convert bytes to ints. Since ints are the preferred type, and to keep the code strict and performant, it would seem like you'd want to make this strictly a `list[int]` (or, IMO, a `Sequence[int]`) so you're not responsible for the conversion (or suffer the penalty of looping over data to convert when there's no need to convert it) and put that on the callers of the API.

Since you're moving to 3.8.x and can allow API changes, this seems like something you could do here, and then retype all of your interfaces to take a `list[int]` and maybe for convenience, provide a helper function for users that used to send `list[bytes]` to convert their list to a `list[int]`.